### ISSUE TITLE
Copy Creating Extra Folders And/Or Failing to Copy Folders on SFTP Remote

### ISSUE THREAD

**Issue by Jdogzz** [ncw/rclone#1274]
When filing an issue, please include the following information if possible as well as a description of the problem.  Make sure you test with the latest beta of rclone.

    http://beta.rclone.org/
    http://rclone.org/downloads/

If you've just got a question or aren't sure if you've found a bug then please use the [rclone forum](https://forum.rclone.org/) instead of filing an issue.

> What is your rclone version (eg output from `rclone -V`)
rclone v1.36-05-g216499dβ
> Which OS you are using and how many bits (eg Windows 7, 64 bit)
Windows 10 64 bit
> Which cloud storage system are you using? (eg Google Drive)
SFTP
> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone.exe copy --verbose "C:\Users\abcd\super long test title" "mysftpremote:super long test title"
> A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
2017/03/23 20:18:41 DEBUG : rclone: Version "v1.36-05-g216499dβ" starting with parameters ["rclone.exe" "copy" "-vv" "C:\\Users\\abcd\\super long test title" "mysftpremote:super long test title"]
2017/03/23 20:18:42 INFO  : sftp://me@sftpaddress:22/super long test title: Modify window is 1s
2017/03/23 20:18:42 INFO  : sftp://me@sftpaddress:22/super long test title: Waiting for checks to finish
2017/03/23 20:18:42 INFO  : sftp://me@sftpaddress:22/super long test title: Waiting for transfers to finish
2017/03/23 20:18:42 INFO  : New Text Document.txt: Copied (new)
2017/03/23 20:18:42 INFO  : another extremely long test title/New Text Document.txt: Copied (new)
2017/03/23 20:18:43 INFO  : even longer super test title/New Rich Text Document.rtf: Copied (new)
2017/03/23 20:18:43 INFO  : even longer super test title/extremely super long test title/New Bitmap Image.bmp: Copied (new)
2017/03/23 20:18:43 INFO  :
Transferred:      7 Bytes (6 Bytes/s)
Errors:                 0
Checks:                 0
Transferred:            4
Elapsed time:        1.1s
2017/03/23 20:18:43 DEBUG : Go routines at exit 11
2017/03/23 20:18:43 DEBUG : rclone: Version "v1.36-05-g216499dβ" finishing with parameters ["rclone.exe" "copy" "-vv" "C:\\Users\\abcd\\super long test title" "mysftpremote:super long test title"]
```

When executing the above command, it creates an extra folder in my sftp remote that has the name "super long test title\even longer super test title" (without the quotes) and is empty. This also occurs in crypt mode for this sftp remote. It appears to correctly upload the folders otherwise, and all subfolders and files in those folders.
I was led to trying this folder arrangement due to another bug I had this morning when uploading a folder structure like the above. For some reason it was fine with uploading the first set of subfolders in the directory, but any folders with subfolders failed (now it seems to be all of them that fail). That error is as follows:
```
2017/03/23 20:33:56 DEBUG : rclone: Version "v1.36-05-g216499dβ" starting with parameters ["rclone.exe" "copy" "-vv" "C:\\Users\\abcd\\Movies1" "mysftpremote:Movies1"]
2017/03/23 20:33:56 INFO  : sftp://me@sftpaddress:22/Movies1: Modify window is 1s
2017/03/23 20:33:56 INFO  : sftp://me@sftpaddress:22/Movies1: Waiting for checks to finish
2017/03/23 20:33:56 INFO  : sftp://me@sftpaddress:22/Movies1: Waiting for transfers to finish
2017/03/23 20:33:57 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Bitmap Image.bmp: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Rich Text Document.rtf: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Text Document.txt: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 ERROR : Attempt 1/3 failed with 3 errors and: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 INFO  : sftp://me@sftpaddress:22/Movies1: Waiting for checks to finish
2017/03/23 20:33:57 INFO  : sftp://me@sftpaddress:22/Movies1: Waiting for transfers to finish
2017/03/23 20:33:57 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Bitmap Image.bmp: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Rich Text Document.rtf: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Text Document.txt: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:57 ERROR : Attempt 2/3 failed with 3 errors and: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:58 INFO  : sftp://me@sftpaddress:22/Movies1: Waiting for checks to finish
2017/03/23 20:33:58 INFO  : sftp://me@sftpaddress:22/Movies1: Waiting for transfers to finish
2017/03/23 20:33:58 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Bitmap Image.bmp: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:58 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Rich Text Document.rtf: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:58 ERROR : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Text Document.txt: Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:58 ERROR : Attempt 3/3 failed with 3 errors and: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
2017/03/23 20:33:58 Failed to copy: Put mkParentDir failed: mkdir "Movies1/Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long" failed: file does not exist
```

I have attached zip files of the folder structure with some blank files that are giving me this error. I do not get any problems when transferring these same directory trees to Google Drive, for example.
[Movies1.zip](https://github.com/ncw/rclone/files/866822/Movies1.zip)
[super long test title.zip](https://github.com/ncw/rclone/files/866823/super.long.test.title.zip)

**Comment by ncw** [comment]
I unpacked the Movies1.zip and then ran

```
$ rclone -vv copy Movies1 TestSftp:Movies2
2017/03/24 10:10:16 DEBUG : rclone: Version "v1.36-05-g216499d" starting with parameters ["rclone" "-vv" "copy" "Movies1" "TestSftp:Movies2"]
2017/03/24 10:10:16 INFO  : sftp://sftptest@localhost:22/Movies2: Modify window is 1s
2017/03/24 10:10:16 INFO  : sftp://sftptest@localhost:22/Movies2: Waiting for checks to finish
2017/03/24 10:10:16 INFO  : sftp://sftptest@localhost:22/Movies2: Waiting for transfers to finish
2017/03/24 10:10:16 INFO  : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Bitmap Image.bmp: Copied (new)
2017/03/24 10:10:16 INFO  : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Text Document.txt: Copied (new)
2017/03/24 10:10:16 INFO  : Trying New Things 1/More Long Trying New Things/Exciting New Things That Are Long/New Rich Text Document.rtf: Copied (new)
2017/03/24 10:10:16 INFO  : 
Transferred:      7 Bytes (31 Bytes/s)
Errors:                 0
Checks:                 0
Transferred:            3
Elapsed time:       200ms
2017/03/24 10:10:16 DEBUG : Go routines at exit 14
2017/03/24 10:10:16 DEBUG : rclone: Version "v1.36-05-g216499d" finishing with parameters ["rclone" "-vv" "copy" "Movies1" "TestSftp:Movies2"]
```

Which didn't replicate the problem...

So it might be Windows specific, or it might be to do with your SFTP server.  Which SFTP server are you using?  Can you get me as many details about that as possible?

**Comment by Jdogzz** [comment]
I just tried transferring the files to the SFTP server (runs CentOS 6.7) from a linux virtual machine (Alpine Linux) and it appears to work perfectly using the newest beta of rclone. I tried again on Windows and it still fails as previously shown (I'm making sure to delete the files it creates each time on the SFTP server so it's fresh). So it does appear to be a Windows specific problem. Is there more information you need supplied to fix this problem?

**Comment by kostix** [comment]
I've just hit the same behaviour while attempting to `rclone` v1.36 amd64 from a Windows 7 machine to a Debian Stretch running OpenSSH.

In my case, the folders `rclone` got confused about weren't insanely long as they were for the OP but they did have Cyrillic names; the names were OK on the remote but the effect was exactly as the OP reported: say, I have a hierarchy which looked like this:

```
Pictures
  Фотографии
      Разобрать
```

and when told to sync `Pictures` "here" to `Pictures` "there",
on the remote side`rclone` created (literally):

```
Pictures
  Фотографии
  Фотографии\Разобрать
```

(and so on for most of nested folders).

Still, it was fine with putting actual picture files located under the `Фотографии` folder to the remote,
and on subsequent runs it had no problem with getting their file info to figure out whether they needed updating.

**Comment by ncw** [comment]
This is almost certainly fixed by a243ea63538fb4598ffb85b971ed9e14826cf0e8

So the latest from https://beta.rclone.org should fix it.

If not, please reopen.

Thanks
